Fixes Secondary ConnectCA update (#17846)
hashicorp/consul: .changelog/17846.txt

```release-note:bug

> connect/ca: Fixes a bug that caused the ConnectCA configuration in secondary DC not persist after initial configuration.

Reviewer: ```suggestion
connect/ca: Fixes a bug preventing CA configuration updates in secondary datacenters
```
